### PR TITLE
Fix component model aot test

### PR DESCRIPTION
**Pull request by LakshanF**
Added some requires types to an rd.xml files, disabled some Jit tests and #73605 fixed some additional tests

### CODE REVIEW

File: src/libraries/System.ComponentModel.TypeConverter/tests/default.rd.xml [hunk 1]
```diff
@@ -0,0 +1,15 @@
+<Directives>
+  <Application>
+    <Assembly Name="mscorlib" />
```
Comment (MichalStrehovsky) [inline src/libraries/System.ComponentModel.TypeConverter/tests/default.rd.xml:3]: I'm a little bit worried that such broadly applied RD.XML will hide product issues. But also I don't know how much of TypeConverter actually is compatible with trimming and whether it even makes sense to guard trim safety.

Just thought to write down what I think.

Comment (LakshanF) [reply src/libraries/System.ComponentModel.TypeConverter/tests/default.rd.xml]: I disabled 7 test scenarios, all in `XTypeDescriptionProviderTests`, to get rid of the rd.xml dependency.

File: src/libraries/System.ComponentModel.TypeConverter/tests/BindableAttributeTests.cs [hunk 2]
```diff
@@ -3,6 +3,7 @@
 
 using System.Collections.Generic;
 using Xunit;
+using System.Diagnostics;
```
Comment (MichalStrehovsky) [inline src/libraries/System.ComponentModel.TypeConverter/tests/BindableAttributeTests.cs:6]:
```suggestion
```

File: src/libraries/tests.proj [hunk 3]
```diff
@@ -420,6 +420,9 @@
     <!-- https://github.com/dotnet/runtime/issues/72908 -->
     <ProjectExclusions Include="$(MSBuildThisFileDirectory)System.Reflection.MetadataLoadContext\tests\System.Reflection.MetadataLoadContext.Tests.csproj" />
 
+    <!-- temp work around to see full test run for https://github.com/dotnet/runtime/pull/73416 -->
+    <ProjectExclusions Include="$(MSBuildThisFileDirectory)System.Resources.ResourceManager\tests\System.Resources.ResourceManager.Tests.csproj" />
+
```
Comment (MichalStrehovsky) [inline src/libraries/tests.proj:425]:
```suggestion
```

**Review by MichalStrehovsky — APPROVED** [review]
Thanks!

File: src/libraries/tests.proj (hunk 3 above)
Comment (MichalStrehovsky) [reply src/libraries/tests.proj:425]: I have a fix for this in #73839. We can rerun after that is merged.

**Pull request merged**